messaginginapp/src/test/java/io/customer/messaginginapp/InAppMessagingStoreTest.kt

```diff
@@ -0,0 +1,372 @@
+    private fun initializeAndSetUser() {
+        manager.dispatch(InAppMessagingAction.Initialize(siteId = String.random, dataCenter = String.random, environment = GistEnvironment.PROD))
+        manager.dispatch(InAppMessagingAction.SetUserIdentifier(String.random))
+    }
+
+    override fun teardown() {
+        manager.dispatch(InAppMessagingAction.Reset)
+        super.teardown()
+    }
+
+    @Test
+    fun givenMessagesWithDifferentPriorities_whenProcessingMessages_thenHighestPriorityMessageIsProcessedFirst() = runTest {
+        val messages = listOf(
+            Message(queueId = "1", priority = 2),
+            Message(queueId = "2", priority = 1),
+            Message(queueId = "3", priority = 3)
+        )
+
+        initializeAndSetUser()
+        manager.dispatch(InAppMessagingAction.ProcessMessageQueue(messages))
+
+        val state = manager.getCurrentState()
+        state.messagesInQueue.size shouldBeEqualTo 3
+        (state.currentMessageState as? MessageState.Processing)?.message?.queueId shouldBeEqualTo "2"
+    }
+
+    @Test
+    fun givenDuplicateMessages_whenProcessingMessages_thenDuplicatesAreRemoved() = runTest {
+        val messages = listOf(
+            Message(queueId = "1"),
+            Message(queueId = "1"),
+            Message(queueId = "2")
+        )
+
+        initializeAndSetUser()
+        manager.dispatch(InAppMessagingAction.ProcessMessageQueue(messages))
+
+        val state = manager.getCurrentState()
+        state.messagesInQueue.size shouldBeEqualTo 2
+        state.messagesInQueue.any { it.queueId == "1" } shouldBe true
+        state.messagesInQueue.any { it.queueId == "2" } shouldBe true
+    }
+
+    @Test
+    fun givenMessageWithSpecificRouteRule_whenRouteChanges_thenMessageStateUpdatesAccordingly() = runTest {
+        initializeAndSetUser()
+        val message = Message(queueId = "1", properties = mapOf("gist" to mapOf("routeRuleAndroid" to "home")))
+
+        manager.dispatch(InAppMessagingAction.ProcessMessageQueue(listOf(message)))
+        manager.dispatch(InAppMessagingAction.SetPageRoute("home"))
+
+        var state = manager.getCurrentState()
+        state.currentRoute shouldBe "home"
+        (state.currentMessageState as? MessageState.Processing)?.message?.queueId shouldBe "1"
+
+        manager.dispatch(InAppMessagingAction.SetPageRoute("profile"))
+
+        state = manager.getCurrentState()
+        state.currentRoute shouldBe "profile"
+
+        val currentState = state.currentMessageState
+        currentState shouldBeInstanceOf MessageState.Dismissed::class.java
+        (currentState as MessageState.Dismissed).message.queueId shouldBe "1"
+    }
+
+    @Test
+    fun givenMultipleMessagesWithDifferentRouteRules_whenRouteChanges_thenCorrectMessageIsDisplayed() = runTest {
+        initializeAndSetUser()
+        val homeMessage = Message(queueId = "1", properties = mapOf("gist" to mapOf("routeRuleAndroid" to "home")))
```

Comment:
Same comment for all `routeRuleAndroid` values in the test, should leverage `pageRuleContains` and `pageRuleEquals`